core/src/main/java/org/apache/metamodel/query/parser/SelectItemParser.java

```diff
@@ -160,6 +164,14 @@ public SelectItem findSelectItem(String expression) throws MultipleSelectItemsPa
                 return new SelectItem(subQuerySelectItem, fromItem);
             }
         }
+        
+        //if the expression is alias of some select item defined return clone of that select item
+        List<SelectItem> allSelectItems = MetaModelHelper.getAllSelectItems(_query);
+        for(SelectItem selectItem : allSelectItems) {
+        	if(selectItem.getAlias() != null && selectItem.getAlias().equals(expressionCopy)) {
+        		return selectItem.clone();
```

Comment:
Do we need to return the clone? Why not return the same select item? If it is referred to by alias (like a kind of pass by reference) then I think it makes sense to keep the same reference item and not a copy.